doc/designs/id-mapping.md

```diff
@@ -158,6 +158,74 @@ In case of explicit allocation, ID range associated with the trusted forest
 Active Directory root domain must have ID range type of `ipa-ad-trust-posix`,
 as well as all other domains visible through the trust link.
 
+## ID range allocation
+
+During deployment FreeIPA takes a random ID range slice from `(1,10000)` and
+multiplies the base offset by 200000. This base becomes a starting point for
+the deployment's ID range.
+
+Both starting point and size of the range can be overridden by an administrator
+who deploys the initial server. The minimal value for the initial ID range
+start cannot be less than `UID_MAX` or `GID_MAX` from `/etc/login.defs`. This
+range is then used by all systems enrolled into the domain managed by this
+initial server.
+
+Thus, a default range that FreeIPA chooses the deployment ID range from is
+200000...2000000000. If this is a single domain and amount of users in it is
+not going to be above 200K, then FreeIPA administrators may never need to
+allocate another ID range within 200000...2000000000 and that original ID range
+stays fixed forever.
+
+There are two other use cases:
+
+ - trusting another IPA or AD deployment, and
+
+ - migrating pre-existing deployment not based on IPA.
+
+When there are multiple organizational domains around and they need to
+interoperate (establish trust between them), additional ID ranges get added on
+the consuming side to represent trusted domain(s). For a trusted domain that is
+an Active Directory, there would be two possibilities, managed within SSSD and
+IPA:
+
+ - an ID range is implicit, defined through a murmurhash3 hash of the domain
+   SID,
+
+ - an ID range is explicit, defined by the admin as UID/GID values stored in AD
+   LDAP user entries for each user
+
+The explicitly defined ID ranges most likely are part of a legacy setup and
```

Comment:
Where are these defined? As in a user-create idrange?

Reply:
No, as I said in a sentence just above this one -- "defined by the admin as UID/GID values stored in AD LDAP user entries for each user". They are explicit by the fact that each relevant user or group entry has UID and GID values in the LDAP attributes in AD LDAP tree.

Reply:
Ok, I guess I'd prefer it to be part of the bullet point since it only applies to that and not to AD in general.